cpp/arcticdb/pipeline/read_options.hpp

```diff
@@ -20,6 +22,7 @@ struct ReadOptions {
     std::optional<bool> set_tz_;
     std::optional<bool> optimise_string_memory_;
     std::optional<bool> batch_throw_on_error_;
+    OutputType output_type_ = OutputType::PANDAS;
```

Comment:
Inconsistent with everything else being optional. I guess they all have implicit defaults, so I guess (not for now) the solution is to remove the optionality from all of those as well.
Unless there are fields where the default value depends on the code path?

Reply:
I don't think they're really the same kind of thing. The rest of the options are tri-state booleans, unset, set-true and set-false, where as you say the default might (potentially) depend on what operation is requested. 

We can prune the `optional<bool>`s, only about half of them are relevant.

I don't think the default output type will end up being used, it's really just here so it isn't left in an uninitialized state. We could potentially have it as an optional and assert that it's set - will have to see what the Python front-end ends up looking like.